File: Bindings~/java/src/main/java/com/alttester/Commands/ObjectCommand/AltSetText.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.ObjectCommand;

import com.alttester.AltObject;
import com.alttester.Commands.AltCommandReturningAltObjects;
import com.alttester.IMessageHandler;

public class AltSetText extends AltCommandReturningAltObjects {

  private AltSetTextParams params;

  public AltSetText(IMessageHandler messageHandler, AltSetTextParams params) {
    super(messageHandler);
    this.params = params;
    params.setCommandName("setText");
  }

  public AltObject Execute() {
    SendCommand(params);
    return ReceiveAltObject(params);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/ObjectCommands/AltSetText.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltSetText : AltCommandReturningAltElement
    {
        AltSetTextParams cmdParams;

        public AltSetText(IDriverCommunication commHandler, AltObject altObject, string text, bool submit) : base(commHandler)
        {
            cmdParams = new AltSetTextParams(altObject, text, submit);
        }

        public AltObject Execute()
        {
            CommHandler.Send(cmdParams);
            return ReceiveAltObject(cmdParams);
        }
    }
}
